src/core/sock/sockinfo_tcp.cpp

```diff
@@ -4556,6 +4556,42 @@ int sockinfo_tcp::tcp_setsockopt(int __level, int __optname, __const void *__opt
             ret = -1;
             errno = EINVAL;
             break;
+        case SO_XLIO_ISOLATE:
+            pass_to_os_cond = false;
+            if (__optlen == sizeof(int)) {
+                if (!m_p_connected_dst_entry && m_rx_ring_map.empty()) {
```

Comment:
these 2 ifs can be combined to reduce nesting. However I would prefer to avoid early break, because currently, all the failure paths are handled at the end when EINVAL is returned and all the success paths finish with a nested break. Handling EINVAL in the early break branch will overcomplicate code in my opinion.

Reply:
combined 2 ifs and added a comment regarding logic of checking dst entry and ring_map